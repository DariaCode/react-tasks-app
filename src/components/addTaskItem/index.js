import AddTaskItem from "./addTaskItem";
export { AddTaskItem };