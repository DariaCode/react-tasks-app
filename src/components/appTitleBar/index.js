import AppTitleBar from "./appTitleBar";
export { AppTitleBar };