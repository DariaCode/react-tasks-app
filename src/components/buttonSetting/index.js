import { ButtonSetting } from './buttonSetting';
export { ButtonSetting }