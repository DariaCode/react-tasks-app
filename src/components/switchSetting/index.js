import { SwitchSetting } from './switchSetting';
export { SwitchSetting }