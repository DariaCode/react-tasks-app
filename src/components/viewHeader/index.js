import { ViewHeader } from "./viewHeader";
export  { ViewHeader };