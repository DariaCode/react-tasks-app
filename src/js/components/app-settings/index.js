import AppSettings from './app-settings';
export { AppSettings }