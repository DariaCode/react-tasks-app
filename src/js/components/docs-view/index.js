import { DocsView } from "./docs-view";
export { DocsView };