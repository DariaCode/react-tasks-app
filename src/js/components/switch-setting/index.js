import { SwitchSetting } from './switch-setting';
export { SwitchSetting }