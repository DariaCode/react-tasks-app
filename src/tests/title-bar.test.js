import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import { TitleBar } from '../js/components/title-bar';

describe('TitleBar', () => {

	it('renders without crashing', () => {
		shallow(<TitleBar />)
	});

});

describe('TitleBar - default render', () => {

	let component;

	beforeAll( () => {
		component = shallow(
			<TitleBar/>
		);
	});

	it('should match snapshot', () => {
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should have the correct class applied to the root node', () => {
		expect(component.find('div').at(0).hasClass('title-bar')).toEqual(true);
	});

	it('should not output left-slot and right-slot containers if not given leftSlot or rightSlot props', () => {
		expect(component.exists('.left-slot')).toEqual(false);
		expect(component.exists('.right-slot')).toEqual(false);
	});

	afterAll( () => {
		component.unmount();
	});

});

describe('TitleBar - props', () => {

	let component;

	beforeAll( () => {
		component = shallow(
			<TitleBar
				classes='test-class'
				leftSlot='Left slot content'
				rightSlot='Right slot content'
			/>
		);
	});

	it('should allow classes to be passed as props to the root node', () => {
		expect(component.find('div').at(0).hasClass('test-class')).toEqual(true);
	});

	it('should output the correct dom node and className if given leftSlot props', () => {
		expect(component.exists('div.left-slot')).toEqual(true);
	});

	it('should output the correct left-slot content if given leftSlot props', () => {
		expect(component.find('div.left-slot').text()).toBe('Left slot content');
	});

	it('should output the correct dom node and className if given rightSlot props', () => {
		expect(component.exists('div.right-slot')).toEqual(true);
	});

	it('should output the correct right-slot content if given rightSlot props', () => {
		expect(component.find('div.right-slot').text()).toBe('Right slot content');
	});

	afterAll( () => {
		component.unmount();
	});

});