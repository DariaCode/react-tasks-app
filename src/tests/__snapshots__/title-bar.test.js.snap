// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar - default render should match snapshot 1`] = `
<div
  className="title-bar "
>
  <div
    className="inner"
  />
</div>
`;
